src/components/ec-dropdown-search/ec-dropdown-search.vue

```diff
@@ -1,435 +1,261 @@
 <template>
-  <div class="ec-dropdown main-menu__header__account-selector main-menu__header__account">
-    <div
-      class="ec-dropdown-search"
-      @blur="show = false"
+  <div
+    ref="popperWidthReference"
+    class="ec-dropdown-search"
+  >
+    <ec-popover
+      v-bind="{
+        open: isOpen,
+        placement: 'bottom',
+        offset: 8,
+        popoverInnerClass: 'ec-dropdown-search__popover',
+        popperOptions,
+        ...popoverOptions,
+      }"
+      @hide="hide"
+      @show="show"
+      @apply-show="focus"
     >
-      <a
-        href
-        class="ec-dropdown-toggle ec-dropdown-toggle--menu"
-        data-toggle="ec-dropdown"
-        aria-haspopup="true"
-        @click.prevent="show = !show"
-      >
-        <span class="title">{{ user.clientName }}</span>
-      </a>
-      <ul
-        ref="scrollbar"
-        v-if="show"
-        class="ec-scrollbar ec-dropdown-menu ec-show"
-      >
-        <li
-          v-if="showSearch"
-          :class="`ec-dropdown-item ec-dropdown-input`"
-        >
-          <i :class="`ec-ico ec-ico-search ec-input-icon`" />
-          <input
-            v-model="searchText"
-            :placeholder="placeholder"
-            @input="hideTooltipsAndAddInteractionHuha()"
-            @focus="addInteractionHuhaTask()"
+      <slot />
+      <div slot="popover">
+        <ul class="ec-dropdown-search__item-list">
+          <li
+            v-if="isSearchEnabled"
+            class="ec-dropdown-search__search-area"
           >
-        </li>
-
-        <li
-          v-for="item in list"
-          :key="item[itemFormat.id]"
-          :class="['ec-dropdown-item ec-ellipsis', addDisabledClass(item)]"
-          :data-id="item[itemFormat.id]"
-          @click="selectItem(item)"
-        >{{ item[itemFormat.text] }}</li>
-      </ul>
-    </div>
+            <ec-icon
+              class="ec-dropdown-search__search-icon"
+              name="simple-search"
+            />
+            <input
+              ref="searchInput"
+              v-model.trim="filterText"
+              autocomplete="off"
+              class="ec-dropdown-search__search-input"
+              :placeholder="placeholder"
+            >
+          </li>
+          <slot
+            name="items"
+            v-bind="filteredItems"
+          >
+            <li
+              v-for="(item, index) of filteredItems"
+              :key="item.id || index"
+              v-ec-tooltip="{
+                placement: 'right',
+                content: item.disabled ? item.disabledReason : '',
+                ...tooltipOptions,
+                ...item.tooltip,
+              }"
+              :title="item.text"
```

Comment:
I think we must discuss this with the design team, because I think for now we only put the tool-tip for the things that are disabled and the once that the text is bigger than the container and then we put 3 dots, but I think it needs to be done with a tool-tip not only with a tittle.

Reply:
We are not going to do the detection for ellipsis (...) in JS, so the title must be there all the time.